src/stravalib/protocol.py

```diff
@@ -74,12 +80,27 @@ def __init__(
             The token that provides access to a specific Strava account.
         requests_session : :class:`requests.Session`
             An existing :class:`requests.Session` object to use.
-
+        rate_limiter : Callable[[dict[str, str], RequestMethod], None], optional
+            A callable used to enforce rate-limiting for API requests.
+            The callable accepts a dict of headers and the
+            HTTP request method as arguments. Defaults to None.
+        token_expires: int
+            Epoch time in seconds when the token expires
+        refresh_token: str
+            Refresh token used to re-authenticate with Strava
+        client_id: int
+            client id for the Strava APP pulled from the users envt
+        client_secret: str
+            client secret for the Strava app pulled from the users envt
         """
         self.log = logging.getLogger(
             "{0.__module__}.{0.__name__}".format(self.__class__)
         )
-        self.access_token = access_token
+        self.access_token: str | None = access_token
+        self.token_expires: int | None = token_expires
+        self.refresh_token: str | None = refresh_token
+        self.client_id: int | None = None
```

Comment:
This two vars as now populated by the method _check_credentials which avoids having to pass things around and perform continual checks!